homeWork2.js

```diff
@@ -0,0 +1,255 @@
+    if (typeof user.id !== "number" || typeof user.name !== "string") return;
+    return user.id !== n && user.id !== m;
+  });
+};
+
+console.log(filterById(users, 1));
+console.log(filterById(users, 1, 2));
+
+////////4///////
+
+const initialArray = [1, "Hello", 3];
+
+const updatedArray = initialArray.map((element, index) => ({
+  elementValue: element,
+  elementType: typeof element,
+  elementIndex: index,
+}));
+
+console.log(updatedArray);
+
+///////5//////
+
+const isPalindrome = (str) => {
+  str = str.toLowerCase();
+  return str === str.split("").reverse().join("");
+};
+
+console.log(isPalindrome("шалаш"));
+
+///////6//////
+
+const newReleases = [
+  {
+    id: 70111470,
+    title: "Die Hard",
+    boxart: "http://cdn-0.nflximg.com/images/2891/DieHard.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+    boxart: "http://cdn-0.nflximg.com/images/2891/BadBoys.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+  {
+    id: 65432445,
+    title: "The Chamber",
+    boxart: "http://cdn-0.nflximg.com/images/2891/TheChamber.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+const updatedNewReleases = newReleases.map((element) => {
+  const { id, title } = element;
+  const newObject = { id, title };
+  return newObject;
+});
+console.log(updatedNewReleases);
+
+////////7/////
+
+const newReleases = [
+  {
+    id: 70111470,
+    title: "Die Hard",
+    boxart: "http://cdn-0.nflximg.com/images/2891/DieHard.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+    boxart: "http://cdn-0.nflximg.com/images/2891/BadBoys.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+  {
+    id: 65432445,
+    title: "The Chamber",
+    boxart: "http://cdn-0.nflximg.com/images/2891/TheChamber.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+const newNewReleases = newReleases.reduce((result, element) => {
+  if (element.rating[0] === 5.0) {
+    const rrr = element.id;
```

Comment:
1) Bad variable name
2) unnecessary variable. You can simplify to result.push(element.id)